messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistModalActivity.kt

```diff
@@ -112,37 +137,40 @@ class GistModalActivity : AppCompatActivity(), GistListener, GistViewListener, T
             animation.setTarget(binding.modalGistViewLayout)
             animation.start()
             animation.doOnEnd {
-                inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("GisModelActivity finish animation completed"))
+                logger.debug("GisModelActivity finish animation completed")
```

Comment:
IIRC, we saw some challenges because dismiss was called, but animation was not complete, causing some unexpected behaviors. Shouldn't we have a state for clarity on being dismissed and being fully dismissed and wiped out for better control?

Reply:
Correct. And IIRC there were some challenges faced with it, probably because we cleared current message reference before animation was complete, so 100ms could be tricky where there were multiple messages and the route changed in-between. Do you think it won't be an issue with new state implementation?